src/mongodb/scores.controller.ts

```diff
@@ -466,224 +474,1089 @@ export class ScoresController {
             } else {
-              prevEle = prevEle + responseTextELE;
-              responseTokenArr.push(prevEle);
+              constructTokenArr.push(constructTextELE);
+              prevEle = constructTextELE;
+              isPrevVowel = false;
             }
-            isPrevVowel = true;
-          } else {
-            responseTokenArr.push(responseTextELE);
-            prevEle = responseTextELE;
-            isPrevVowel = false;
           }
         }
-      }
 
 
-      // Comparison Logic
+        // Comparison Logic
 
-      for (let originalTokenArrEle of originalTokenArr) {
-        if (responseTokenArr.includes(originalTokenArrEle)) {
-          correctTokens.push(originalTokenArrEle);
-        } else {
-          missingTokens.push(originalTokenArrEle);
+        for (let originalTokenArrEle of originalTokenArr) {
+          if (constructTokenArr.includes(originalTokenArrEle)) {
+            correctTokens.push(originalTokenArrEle);
+          } else {
+            missingTokens.push(originalTokenArrEle);
+          }
         }
-      }
 
-      let filteredTokenArr = [];
+        let missingTokenSet = new Set(missingTokens);
 
-      //token list for ai4bharat response
-      let tokenArr = [];
-      let anamolyTokenArr = [];
+        missingTokens = Array.from(missingTokenSet)
 
-      // Create Single Array from AI4bharat tokens array
-      createScoreDto.output[0].nBestTokens.forEach(element => {
-        element.tokens.forEach(token => {
-          let key = Object.keys(token)[0];
-          let value = Object.values(token)[0];
+        let filteredTokenArr = [];
 
-          let insertObj = {};
-          insertObj[key] = value;
-          tokenArr.push(insertObj);
+        //token list for ai4bharat response
+        let tokenArr = [];
+        let anamolyTokenArr = [];
 
-          let key1 = Object.keys(token)[1];
-          let value1 = Object.values(token)[1];
-          insertObj = {}
-          insertObj[key1] = value1;
-          anamolyTokenArr.push(insertObj);
+        // Create Single Array from AI4bharat tokens array
+        CreateLearnerProfileDto.output[0].nBestTokens.forEach(element => {
+          element.tokens.forEach(token => {
+            let key = Object.keys(token)[0];
+            let value = Object.values(token)[0];
+
+            let insertObj = {};
+            insertObj[key] = value;
+            tokenArr.push(insertObj);
+
+            let key1 = Object.keys(token)[1];
+            let value1 = Object.values(token)[1];
+            insertObj = {}
+            insertObj[key1] = value1;
+            anamolyTokenArr.push(insertObj);
+          });
         });
-      });
 
-      let uniqueChar = new Set();
-      prevEle = '';
-      isPrevVowel = false;
+        let uniqueChar = new Set();
+        prevEle = '';
+        isPrevVowel = false;
 
-      // Create Unique token array
-      for (let tokenArrEle of tokenArr) {
-        let tokenString = Object.keys(tokenArrEle)[0];
```

Comment:
we may use the transformer library to avoid these complex unicode construction, comparisons.  Or should be moved as separate library and use them here, to keep the code lines minimal